### DISCUSSION TITLE
New build dependencies

### DISCUSSION THREAD

**Discussion by alxpl** [highperformancecoder/scidavis-old#212] [General]
Hello everyone,

It's been a while since I last built a development snapshot of SciDAVis and there have been a lot of changes since the 2.3.0 release (which is really good for the project). I think I might find a couple of hours to run some tests this weekend (as it turns out PhDs really do take up a lot of one's time), so that I don't botch the Fedora updates whenever a new release appears. Is anyone using a build system that builds rpms and contains recent builds? Unless I'm looking at the wrong place, the most recent rpm-related change in Russel's [OBS project page](https://build.opensuse.org/package/show/openSUSE:Factory/scidavis) is from 8 months ago.

Alternatively, could someone take a look at the current dependencies in my spec file and tell me off the top of their head what is obsolete?

These are the dependencies of the main package:
```
BuildRequires:  make
BuildRequires:  gcc-c++
BuildRequires:  python3
BuildRequires:  desktop-file-utils
BuildRequires:  gsl-devel
BuildRequires:  liborigin-devel
BuildRequires:  mesa-libGLU-devel
BuildRequires:  muParser-devel
BuildRequires:  qt5-qtbase-devel
BuildRequires:  qwt5-qt5-devel
BuildRequires:  qwtplot3d-qt5-devel
BuildRequires:  python3-sip-devel
BuildRequires:  python3-pyqt5-sip
BuildRequires:  qt5-qtsvg-devel
BuildRequires:  qt5-linguist
BuildRequires:  zlib-devel
BuildRequires:  libappstream-glib
# required for the tests, enable when building locally
#BuildRequires:  xorg-x11-server-Xvfb
#BuildRequires:  unittest-cpp-devel
#BuildRequires:  boost-devel

Requires:       python3-qt5
Requires:       hicolor-icon-theme
Requires:       kde-filesystem

Recommends:     python3-%{name}
```

the python3 module depends on python3-devel and python3-qt5-devel, but as far as I can tell, these won't require any changes.

If nobody can spare the cycles, that's ok, I think I'll figure things out by myself eventually.

All the best,
A.

**Comment by highperformancecoder** [comment]
On Wed, Mar 24, 2021 at 01:11:33PM -0700, alxpl wrote:
> Hello everyone,
> 
> It's been a while since I last built a development snapshot of SciDAVis and
> there have been a lot of changes since the 2.3.0 release (which is really good
> for the project). I think I might find a couple of hours to run some tests this
> weekend (as it turns out PhDs really do take up a lot of one's time), so that I
> don't botch the Fedora updates whenever a new release appears. Is anyone using
> a build system that builds rpms and contains recent builds? Unless I'm looking
> at the wrong place, the most recent rpm-related change in Russel's OBS project
> page is from 8 months ago.

Yeah, no sorry - I haven't updated the OBS project for the new cmake
build process.

At the moment, I am stuck on getting the MacOSX build to work - it
works on a developer machine, but as soon as I rewrite the executables
and dylibs to refer to the locally packages dylibs, it core dumps,
right when opening the QtSettings file.

> 
> Alternatively, could someone take a look at the current dependencies in my spec
> file and tell me off the top of their head what is obsolete?
>

Basically you should remove liborigin-devel, qwt5-qt5-devel and
qwtplot3d-qt5-devel, and instead using the git submodule system to
build and include the pacthed versions of these libraries. This works
with the cmake system - which is the way forward. Building by qmake is
like to break any time soon, if not already broken.

I think I updated the building page on github with instructions how to do that.

> These are the dependencies of the main package:
> 
> BuildRequires:  make
> BuildRequires:  gcc-c++
> BuildRequires:  python3
> BuildRequires:  desktop-file-utils
> BuildRequires:  gsl-devel
> BuildRequires:  liborigin-devel
> BuildRequires:  mesa-libGLU-devel
> BuildRequires:  muParser-devel
> BuildRequires:  qt5-qtbase-devel
> BuildRequires:  qwt5-qt5-devel
> BuildRequires:  qwtplot3d-qt5-devel
> BuildRequires:  python3-sip-devel
> BuildRequires:  python3-pyqt5-sip
> BuildRequires:  qt5-qtsvg-devel
> BuildRequires:  qt5-linguist
> BuildRequires:  zlib-devel
> BuildRequires:  libappstream-glib
> # required for the tests, enable when building locally
> #BuildRequires:  xorg-x11-server-Xvfb
> #BuildRequires:  unittest-cpp-devel
> #BuildRequires:  boost-devel
> 
> Requires:       python3-qt5
> Requires:       hicolor-icon-theme
> Requires:       kde-filesystem
> 
> Recommends:     python3-%{name}
> 
> the python3 module depends on python3-devel and python3-qt5-devel, but as far
> as I can tell, these won't require any changes.
> 
> If nobody can spare the cycles, that's ok, I think I'll figure things out by
> myself eventually.
> 
> All the best,
> A.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Reply by alxpl** [reply]
Hello Russel,

Thanks for the update.

> Basically you should remove liborigin-devel, qwt5-qt5-devel and
> qwtplot3d-qt5-devel, and instead using the git submodule system to
> build and include the pacthed versions of these libraries. This works
> with the cmake system - which is the way forward. Building by qmake is
> like to break any time soon, if not already broken.

I don't think that's going to work in Fedora and probably other distributions, unless I don't fully grasp how submodules work. As far as I can tell our build systems do not get internet access during build time and everything has to be declared in the spec file (and be part of the distribution). Bundling libraries, though no longer completely forbidden, is still discouraged. Besides, I'm the also the maintainer of two out of these three packages, so it doesn't complicate things any further for me at least, to update all the packages at once. However, modifying them with SciDAVis specific code is a whole different beast…




> 
> I think I updated the building page on github with instructions how to do that.

[This](https://github.com/highperformancecoder/scidavis/blob/master/INSTALL.md) one?

**Comment by highperformancecoder** [comment]
On Wed, Mar 24, 2021 at 05:15:32PM -0700, alxpl wrote:
> Hello Russel,
> 
> Thanks for the update.
> 
>     Basically you should remove liborigin-devel, qwt5-qt5-devel and
>     qwtplot3d-qt5-devel, and instead using the git submodule system to
>     build and include the pacthed versions of these libraries. This works
>     with the cmake system - which is the way forward. Building by qmake is
>     like to break any time soon, if not already broken.
> 
> I don't think that's going to work in Fedora and probably other distributions,
> unless I don't fully grasp how submodules work. As far as I can tell our build
> systems do not get internet access during build time and everything has to be
> declared in the spec file (and be part of the distribution). Bundling
> libraries, though no longer completely forbidden, is still discouraged.
> Besides, I'm the also the maintainer of two out of these three packages, so it
> doesn't complicate things any further for me at least, to update all the
> packages at once. However, modifying them with SciDAVis specific code is a
> whole different beast…

a) SciDAVis can be built against system installed libraries via CMAKE options
b) The tarballs for official releases will contain the checked-out submodule directories, when I finally get around to finishing a release.

Both liborigin and qwt5-qt5 are Miquel Agueda's projects, and
considerably forked from mainstream qwt, and maybe liborigin too. But
even those repos have been forked for SciDAVis specific fixes. You
could merge upstream, I suppose, but its a lot of extra work, and
particularly the qwt5-qt5 and qwplot3d-qt5 projects are really just for
supporting SciDAVis only, AFAICT.

> 
>     I think I updated the building page on github with instructions how to do
>     that.
> 
> This one?
>

Don't know - there's no URL here...



-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Reply by alxpl** [reply]
> a) SciDAVis can be built against system installed libraries via CMAKE options

That's good to know. As long as it's only the build time dependencies, there's some wiggle room to get things working.

Miguel has been contributing directly to liborigin upstream and given Qwt's development state, it was relatively easy to get the qwt5-qt5 fork accepted and packaged in Fedora. I may have to bother qwplot3d's maintainer at some point, but everything seems doable so far.

I was asking about this page:
https://github.com/highperformancecoder/scidavis/blob/master/INSTALL.md

**Reply by Suthiro** [reply]
> Both liborigin and qwt5-qt5 are Miquel Agueda's projects, and considerably forked from mainstream qwt, and maybe liborigin too. But even those repos have been forked for SciDAVis specific fixes.

I just checked and found that there is almost no difference between scidavis/liborigin and [liborigin upstream](https://sourceforge.net/p/liborigin/git/ci/master/tree/). Aside cmake related changes, only 9 (nine) lines of code were changed. The changes are purely (arguably minor) bugfixes. So I wonder do we really need our separate liborigin submodule?
Also, I just realized that we apparently do not use liborigin2 at all, but have it for some reason in `3rdparty`.

**Reply by alxpl** [reply]
I suppose Miquel can push any bugfixes upstream, so everything will be in sync. Besides, as far as I remember, a lot of changes to liborigin's code base were made with SciDAVis in mind.

I don't think there's any reason to use liborigin2, I've retired it in Fedora.

**Reply by Suthiro** [reply]
liborigin2 was removed in #213 without any changes to SciDAVis' codebase. I vote for unification with liborigin-upstream.

**Comment by highperformancecoder** [comment]
On Thu, Mar 25, 2021 at 12:52:42AM -0700, alxpl wrote:
>     a) SciDAVis can be built against system installed libraries via CMAKE
>     options
> 
> That's good to know. As long as it's only the build time dependencies, there's
> some wiggle room to get things working.
> 
> Miguel has been contributing directly to liborigin upstream and given Qwt's
> development state, it was relatively easy to get the qwt5-qt5 fork accepted and
> packaged in Fedora. I may have to bother qwplot3d's maintainer at some point,
> but everything seems doable so far.
> 
> I was asking about this page:
> https://github.com/highperformancecoder/scidavis/blob/master/INSTALL.md

Ah, yes - that still refers to the old qmake method. It needs to be updated, but cmake is a pretty simple system to find your way around. Any questions, please post here - there's a few people around bulding from source code.

> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by alxpl** [comment]
Earlier today Stefan Gerlach merged the commits in the liborigin submodule with the upstream repository, so there's no need for it right now. Since there are other consumers of liborigin, such as LabPlot, it would be a good idea to push changes directly upstream and avoid needless forking and fragmentation.

Last night I built the latest snapshot of qwt5-qt5, tonight I'm going to work on liborigin and SciDAVis, so hopefully sometime tomorrow I will submit the 2.4.0 update to Fedora repositories.

**Comment by highperformancecoder** [comment]
Sigh - I see what you mean. I thought these repos that Chinoune had
been github forked, so that it would make it easier to issue pull
requests on the the original repo.

But I now see they've been forked forked, with not connection to the original.

Could you please tell me what are the official forks of liborigin,
qwt5-qt5 and qwtplot3d, and I'll try to see if I can establish a
connection to allow pushing changes upstream.

Cheers

On Sat, May 08, 2021 at 01:25:57PM -0700, alxpl wrote:
> Earlier today Stefan Gerlach merged the commits in the liborigin submodule with
> the upstream repository, so there's no need for it right now. Since there are
> other consumers of liborigin, such as LabPlot, it would be a good idea to push
> changes directly upstream and avoid needless forking and fragmentation.
> 
> Last night I built the latest snapshot of qwt5-qt5, tonight I'm going to work
> on liborigin and SciDAVis, so hopefully sometime tomorrow I will submit the
> 2.4.0 update to Fedora repositories.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Reply by alxpl** [reply]
* liborigin:
https://sourceforge.net/projects/liborigin/
Miquel and Stefan are the people working on it at the moment and its development has been following that of SciDAVis and LabPlot.

* qwt5-qt5 was forked off the last maintenance release of the Qwt 5.x branch, 5.2.3, by Miquel. I've seen other distributions package it as well, mostly in order to build SciDAVis, so until there's a migration to a newer version of Qwt, for all intents and purposes, Miquel's repo is the upstream source:
https://github.com/gbm19/qwt5-qt5

* qwtplot3d hasn't seen an update since 2007 and it has been a pain for package maintainers to keep around and build with Qt5. It's download page lists a 0.3.0 version, but there's nothing more recent than 0.2.7:
http://qwtplot3d.sourceforge.net/
I've seen people replace it with functionality from Qt's own Qt Data Visualization module:
https://doc.qt.io/qt-5/qtdatavisualization-index.html
That includes a fork of SciDAVis, AlphaPlot:
https://github.com/narunlifescience/AlphaPlot/commit/68780915e2ff4846861d7689167363468c8af01d
In Fedora, the only other package that depends on qwtplot3d is [COPASI](http://copasi.org/) and it looks like they are (very slowly) moving away from it too.

**Reply by Suthiro** [reply]
Right now it is not possible to produce publication-ready graphs using SciDAVis. Huge effort is needed to catch up to modern plotting libraries (e.g. matplotlib and gnuplot), the only reason to keep native plotting is the possibility of data analyzing (i.e. interactive graphs bound to data). I've started porting the codebase to qwt6 earlier this year, but there is no free time recently to finish the job, sorry. So I think we should drop native 3d plotting (and therefore qwtplot3d)  since I doubt it that anyone uses this functionality at all.
P.S. Tried the latest build of AlphaPlot and it crashed almost immediately. Lol

**Reply by narunlifescience** [reply]
@Suthiro 3d plotting is still experimental in Alphaplot.

**Reply by Suthiro** [reply]
strongly off-topic, but
@narunlifescience 
FYI the app crashes randomly on startup (approx each 10th launch, some race conditions?) and always while closing with shared violation at 0xc0000005. To reproduce: launch Alphaplot , close Alphaplot . Exception is thrown. Win7 x64 & Alphaplot 1.011. I've not even tried to plot anything.

**Reply by narunlifescience** [reply]
This is most likely fixed in changes post release. unfortunately I don't have an windows executable of the same to provide. However If you have access to Linux, you can try the developer version with Flathub Beta.
https://github.com/narunlifescience/AlphaPlot/issues/20

**Reply by alxpl** [reply]
> Right now it is not possible to produce publication-ready graphs using SciDAVis. Huge effort is needed to catch up to modern plotting libraries (e.g. matplotlib and gnuplot), the only reason to keep native plotting is the possibility of data analyzing (i.e. interactive graphs bound to data). I've started porting the codebase to qwt6 earlier this year, but there is no free time recently to finish the job, sorry. So I think we should drop native 3d plotting (and therefore qwtplot3d) since I doubt it that anyone uses this functionality at all.

I hope I didn't come off as taking jabs at anyone, hence no need to apologize for not having enough time to volunteer. I was merely laying out the facts and adding the POV of a distribution maintainer.

**Reply by Suthiro** [reply]
@narunlifescience 
may I ask you why you decided to fork?

@alxpl no-no, I feel guilty because I don't like to leave something unfinished, even if the task was assigned to me by myself :)

**Reply by narunlifescience** [reply]
@Suthiro I was in need of a graphing software which is quite easy to use like scidavis but can generate publication quality graphs. One can ask why you didn't try improving the software instead of forking it. Simple answer is SciDavis has a huge user base which means you cannot suddenly disrupt/change major features in the software and break things. So I started Alphaplot as an experiment to do the same.

Accidentally some one stumbled up on the fork and asked for a release. I cautioned him multiple times that the software is no different from SciDavis at that time point. Finally I released a version which was more or less same as its parent software. After that I was in the process of porting to QT5 and other major library changes. If you look at the disruptive changes that AlphaPlot made with respect to the code base of SciDavis, it is quite evident that the compatibility with previous versions are broken at multiple levels. So forking was the only option. Alpha plot is still not perfect or even close but I am happy with some features like property manager, the way it interact with graphs (especially the new layout management system) and the quality of graph it produces. 
May be at some time point in future we can think of working together for the same goal.

**Reply by Suthiro** [reply]
@narunlifescience thank you for answering! `QCustomPlot` is a great library, it founds usage it in my projects extensively. Do I understand right that you have dropped qwt/qwtplot3d completely?

**Reply by narunlifescience** [reply]
@Suthiro yes, that library is a relic now. I shifted to QtDataVisualization library just to give the application some basic 3d plotting capabilities. As you pointed out earlier, only a few users are using 3D plotting capabilities. So I thought this basic library would be good enough.

**Comment by highperformancecoder** [comment]
Thanks for the info. I will look at these, and attempt to "derive" the
versions used by SciDAVis from those version, to enable pushing
changes upstream. For liborigin and qwt-qt5 at least - qwtplot3d looks
like it is abandonware - I don't think we can do anything here.

Cheers

On Sun, May 09, 2021 at 03:45:00AM -0700, alxpl wrote:
>   • liborigin:
>     https://sourceforge.net/projects/liborigin/
>     Miquel and Stefan are the people working on it at the moment and its
>     development has been following that of SciDAVis and LabPlot.
> 
>   • qwt5-qt5 was forked off the last maintenance release of the Qwt 5.x branch,
>     5.2.3, by Miquel. I've seen other distributions package it as well, mostly
>     in order to build SciDAVis, so until there's a migration to a newer version
>     of Qwt, for all intents and purposes, Miquel's repo is the upstream source:
>     https://github.com/gbm19/qwt5-qt5
> 
>   • qwtplot3d hasn't seen an update since 2007 and it has been a pain for
>     package maintainers to keep around and build with Qt5. It's download page
>     lists a 0.3.0 version, but there's nothing more recent than 0.2.7:
>     http://qwtplot3d.sourceforge.net/
>     I've seen people replace it with functionality from Qt's own Qt Data
>     Visualization module:
>     https://doc.qt.io/qt-5/qtdatavisualization-index.html
>     That includes a fork of SciDAVis, AlphaPlot:
>     ***@***.***
>     In Fedora, the only other package that depends on qwtplot3d is COPASI and
>     it looks like they are (very slowly) moving away from it too.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by highperformancecoder** [comment]
If we can avoid the move to Qwt6 by using this new library, lets go for it.

I already have had one attempt at Qwt6 migration, and I know it is a
big and horrible job, given just how many C++ conventions are broken
by that library.

On Mon, May 10, 2021 at 12:33:44AM -0700, Suthiro wrote:
> @narunlifescience thank you for answering! QCustomPlot is a great library, it
> founds usage it in my projects extensively. Do I understand right that you have
> dropped qwt/qwtplot3d completely?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Reply by Suthiro** [reply]
It looks like the reintegration of the transition to QCustomPlot from Alphaplot to SciDAVis' codebase will be harder than migrating to qwt6. Maybe someone who mastered git is able to merge codebases in right way. So we will need either to start almost from scratch with QCustomPlot or try to move to qwt6. I would stick to the latter.
I also promised try to build MSVC + Python bundle, and I report that I'm not able. Microsoft compiler throws a lot of syntax errors in sip-generated files. I wonder how it works in github CI. Also, is there any reason that Python is disabled for Qt 5.12 and enabled for Qt 5.15?

**Reply by narunlifescience** [reply]
@highperformancecoder @Suthiro 
yes, I agree. Why don't we work on AlphaPlot and make it reverse compatible with scidavis files? that should be far more easy to do. why don't we make Alphaplot a new version of scidavis? Considering both the projects lack active developers, I feel this is a wise option to consider.

Any way it is up to the developers to decide on this. of course each project will have its own future prospects and developer opinions to consider.

**Reply by Suthiro** [reply]
@narunlifescience I'm sure that it is much simpler to integrate changes from AlphaPlot to SciDAVis than vice versa. Also, SciDAVis already has support for SciDAVis files, and, as you stated, SciDAVis has large user base and AlphaPlot doesn't. Not to mention that SciDAVis is updated more or less regular and has more active developers right now. So it would be really great if you would reintegrate your amazing efforts into upstream. I could help with that.

**Comment by highperformancecoder** [comment]
On Fri, May 14, 2021 at 12:25:57PM -0700, Suthiro wrote:
> @narunlifescience I'm sure that it is much simpler to integrate changes from
> AlphaPlot to SciDAVis than vice versa. Also, SciDAVis already has support for
> SciDAVis files, and, as you stated, SciDAVis has large user base and AlphaPlot
> doesn't. Not to mention that SciDAVis is updated more or less regular and has
> more active developers right now. So it would be really great if you would
> reintegrate your amazing efforts into upstream. I could help with that.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

From my perspective, I responded to Tilman Benkert's request for
someone to take over SciDAVis thinking that it might help me brush up
my Qt chops. To a certain extent that has happened, but things have
panned out quite differently, and I'm now unlikely to really do much
Qt work other than on SciDAVis. I'm much more likely to need
Javascript UI skills in future projects.

I've also not really found a use for SciDAVis for my own plotting
needs - it lacks in publication quality output, and also does not
scale well to large datasets. I have found I continue to use and
invest in GNUplot, which being command line driven, performs extremely
well on those points. The former issue (publication quality output)
could be easily enough addressed, of course, the latter about dataset
size scalability can never be - it is a limitation to having a
spreadsheet built into a plotting library.

Nevertheless, I'm committed as ever to not letting SciDAVis "wither on
the vine". It has proved an interesting exercise in managing a
moderately large legacy codebase, and in managing an open source
multinational community.

As for comments on whether AlphaPlot should be merged into SciDAVis or
AlphaPlot become the new SciDAVis, I am quite open to either
possibility. SciDAVis does have quite a large user community, and that
has value. But the value really is in the brand name, not the code,
which is large, crufty (due in no small part to a failed "aspect
oriented" redesign) and relying on out-of-date libraries. It was quite
a struggle to get it rejigged for Python 3 and Qt5 (still not quite
there on Macs), and there is substantial effort required to move to
Qwt6.

Cheers

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     ***@***.***
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Reply by Suthiro** [reply]
@highperformancecoder 

>I've also not really found a use for SciDAVis for my own plotting
needs - it lacks in publication quality output, and also does not
scale well to large datasets. I have found I continue to use and
invest in GNUplot, which being command line driven, performs extremely
well on those points. The former issue (publication quality output)
could be easily enough addressed, of course, the latter about dataset
size scalability can never be - it is a limitation to having a
spreadsheet built into a plotting library.

How large datasets are you using? Recently ~500 Mb dataset was successfully processed it quite quick using modern but average rig. Concerning the plotting quality, I'll repeat [myself](https://sourceforge.net/p/scidavis/scidavis-bugs/414/#cb60/de04):
"Actually, I consider SciDAVis as a place to keep my data and scripts together, so I rarely use fit wizard or fancy plots, because there are scipy and gnuplot. IMO even Origin is far inferior to these two combined. SciDAVis allows one to use full-scale python scripting, so I prepare fitting scripts in Spyder and then just paste 'em into the project(s). Works well.
...
PS I'm working on gnuplot integration right now. Hopefully it would be usable (and useful)."

Hopefully, there will be some free time in summer.

>Nevertheless, I'm committed as ever to not letting SciDAVis "wither on
the vine". It has proved an interesting exercise in managing a
moderately large legacy codebase, and in managing an open source
multinational community.

Thank you very much for that. From my point of view, you saved SciDAVis from obsoleting.
